Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,103 @@
+# Kubernetes On-Premises Deployment From Scratch -- Start to Finish #
+This guide will walk you through deploying *Kubernetes 1.8* on a Linux master and join two Windows nodes to it without a cloud provider.
+
+**Note**: The guide will assume a local working directory of `~/kube` for the Kubernetes setup. If you choose a different directory, just replace any references to that path.
+
+## Building the Binaries ##
+We will begin with a Linux master node and a Windows worker node on RS3+, both of which are assumed to be VMs.
+
+First, install all of the pre-requisites on the master:
+
+    # apt-get install git build-essentials docker.io
+
+### Building Kubernetes ###
+We will need to build the `kubelet` and `kubeproxy` binaries for Windows from scratch _cross-compiling from Linux_, so let's [set up a Go environment](https://golang.org/doc/install#install) for that, first. This guide assumes Go 1.9 and has not been tested with anything else. With your Go environment ready (don't forget to set your `$GOPATH`!), we can build the Windows binaries:
+
+```bash
+$ K8SREPO="github.com/madhanrm/kubernetes"
+$ go get -d $K8SREPO
+$ cd $GOPATH/src/$K8SREPO
+$ KUBE_BUILD_PLATFORM=windows/amd64 make WHAT=cmd/kubectl
+$ cp ~/_output/bin/kubectl.exe ~/kube/
+$ git checkout cniwindows
+$ make clean && KUBE_BUILD_PLATFORM=windows/amd64 make WHAT=cmd/kubelet
+$ cp ~/_output/bin/kubelet.exe ~/kube/

```

Comment:
There seems to be a missing step here. In my environment, I had to execute the 'make' for kubelet twice (once for Linux and once for Windows). If I don't first build for Linux, then commands like "defaulter-gen" won't be present in _output/bin and the Windows 'make' will fail. This seems to be what Madhan was indicating in his OneNote steps...it should be like this:

```
make clean && make WHAT=cmd/kubelet
KUBE_BUILD_PLATFORMS=windows/amd64 make WHAT=cmd/kubelet
```

Reply:
Also, just like Madhan, my output binary was in: _output/local/bin/windows/amd64/kubelet.exe

Reply:
Ah, that's interesting. This may mean people won't need to fiddle with the italicized "if you get permission errors" step. I'll try this out to confirm and edit the guide to reflect that. Good tip!